partiql-lang/src/main/kotlin/org/partiql/lang/eval/visitors/SubqueryCoercionVisitorTransform.kt

```diff
@@ -0,0 +1,153 @@
+            is PartiqlAst.Expr.Modulo -> n.copy(operands = n.operands.map { coerceToSingle(it) })
+            is PartiqlAst.Expr.Concat -> n.copy(operands = n.operands.map { coerceToSingle(it) })
+            is PartiqlAst.Expr.And -> n.copy(operands = n.operands.map { coerceToSingle(it) })
+            is PartiqlAst.Expr.Or -> n.copy(operands = n.operands.map { coerceToSingle(it) })
+
+            is PartiqlAst.Expr.Eq -> n.copy(operands = coerceInComparisonOps(n.operands))
+            is PartiqlAst.Expr.Ne -> n.copy(operands = coerceInComparisonOps(n.operands))
+            is PartiqlAst.Expr.Gt -> n.copy(operands = coerceInComparisonOps(n.operands))
+            is PartiqlAst.Expr.Gte -> n.copy(operands = coerceInComparisonOps(n.operands))
+            is PartiqlAst.Expr.Lt -> n.copy(operands = coerceInComparisonOps(n.operands))
+            is PartiqlAst.Expr.Lte -> n.copy(operands = coerceInComparisonOps(n.operands))
+
+            is PartiqlAst.Expr.Like -> n.copy(value = coerceToSingle(n.value), pattern = coerceToSingle(n.pattern))
+            is PartiqlAst.Expr.Between -> n.copy(value = coerceToSingle(n.value), from = coerceToSingle(n.from), to = coerceToSingle(n.to))
+
+            is PartiqlAst.Expr.InCollection -> toSingleInInCollection(n)
+
+            // TODO Revisit the following expression forms, where this implementation does not coerce most potential SELECT subqueries,
+            // being at odds with the literal letter of the Ch 9 of the PartiQL specification.
+            // Some of these are clear-cut cases, others need to be considered and decided upon.
+            is PartiqlAst.Expr.IsType -> n
+            is PartiqlAst.Expr.SimpleCase -> n
+            is PartiqlAst.Expr.SearchedCase -> n
+            is PartiqlAst.Expr.Struct -> n
+            is PartiqlAst.Expr.Bag -> n
+            is PartiqlAst.Expr.List -> n
+            is PartiqlAst.Expr.Sexp -> n
+            is PartiqlAst.Expr.Date -> n
+            is PartiqlAst.Expr.LitTime -> n
+            is PartiqlAst.Expr.BagOp -> n
+
+            is PartiqlAst.Expr.GraphMatch -> n.copy(expr = coerceToSingle(n.expr))
+
+            // ... continuing the expressions with so far un-coerced subqueries
+            is PartiqlAst.Expr.Path -> n
+            is PartiqlAst.Expr.Call -> n
+            is PartiqlAst.Expr.CallAgg -> n
+            is PartiqlAst.Expr.CallWindow -> n
+            is PartiqlAst.Expr.Cast -> n
+            is PartiqlAst.Expr.CanCast -> n
+            is PartiqlAst.Expr.CanLosslessCast -> n
+            is PartiqlAst.Expr.NullIf -> n
+            is PartiqlAst.Expr.Coalesce -> n
+
+            is PartiqlAst.Expr.Select ->
+                n.copy(
+                    setq = n.setq,
+                    project = toSingleInProject(n.project),
+                    from = n.from, // per SQL, don't coerce subqueries that are data sources in FROM
+                    fromLet = n.fromLet, // LET is PartiQL-specific, so allow binding to a non-scalar
```

Comment:
> (Aside: from here one can proceed musing whether LET could be more useful as an AT-like modifier on any participant in a join in FROM, rather than a SELECT clause that follows FROM...)

Not a blocker for this PR: let's discuss more about this as this seems to be an interesting insight.